### PR TITLE
Support lumi-weighted input files for the TkAl Primary Vertex Validation plotting

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The title tells it all. It was recently realized that some of the plots that were made public in [CMS-DP-2024-071](https://twiki.cern.ch/twiki/bin/view/CMSPublic/TkAlignmentPerformanceRun3Reprocessing) are affected by a bug in the computation of the error bars.
Upon further investigation an issue was found with the error bars in this particular set of plots ([primary vertex run averaged](https://twiki.cern.ch/twiki/bin/view/CMSPublic/TkAlignmentPerformanceRun3Reprocessing#Primary_vertex_validation_Averag)): in general we use a likelihood gaussian fit to account for empty bins when we look at run by run primary vertex validation during operations which can happen relatively frequently in single runs. However, this is not appropriate for weighted inputs (as is the case for lumi-averaging) and here one should use a least squares fit. 
This PR provides some changes to plotting macro `FitPVResiduals` such that it auto-detects if the input file is lumi-weighted or a regular single run file and then acts accordingly in the fit procedure.
Some preliminary commit (86e2e0d073610be6b1fa2f8fd1d1b1874cf98fa3) was needed in order to make the fetching of the validation "metadata" agnostic on the type of input file (before this PR this was dealt with by using a separate and different code).

#### PR validation:

I've run this branch with and without the commit acb0ab0ddcb937aaab58d320d0918a5f46e665a7. 
Without the changes I get plot with exaggerated error bars e.g:

image

while, when using the commit I get much more reasonable error bars:

image

In addition I have also tested the single run (unweighted case) to make sure there are no changes, and none are observe (see [this](https://marcomusich.web.cern.ch/TestPVValidation/single/mods/) vs [that](https://marcomusich.web.cern.ch/TestPVValidation/single/vanilla/)).

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

Cc:
@henriettepetersen @TomasKello

### CODE REVIEW

**Comment by mmusich** [comment]
type bug-fix

File: Alignment/OfflineValidation/macros/FitPVResiduals.C [hunk 1]
```diff
@@ -698,10 +727,10 @@ void FitPVResiduals(TString namesandlabels,
 
     if (gDirectory->GetListOfKeys()->Contains("pTinfo")) {
       gDirectory->GetObject("pTinfo", thePtInfoHistos[i]);
-      thePTBINS[i] = thePtInfoHistos[i]->GetBinContent(1) * 3. / thePtInfoHistos[i]->GetEntries();
```
Comment (perrotta) [inline Alignment/OfflineValidation/macros/FitPVResiduals.C:701]: What was the original purpose of multiplying the bin content by 3?

Comment (mmusich) [reply Alignment/OfflineValidation/macros/FitPVResiduals.C:701]: 3 bins in the input histogram.

**Comment by perrotta** [comment]
+alca

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**